crates/bevy_ecs/src/component/mod.rs

```diff
@@ -24,7 +25,26 @@ use thiserror::Error;
 ///
 /// Components can be grouped together into a [`Bundle`](crate::bundle::Bundle).
 pub trait Component: Send + Sync + 'static {}
-impl<T: Send + Sync + 'static> Component for T {}
+
+// ECS dependencies cannot derive Component, so we must implement it manually for relevant structs.
+impl<T> Component for bevy_tasks::Task<T> where Self: Send + Sync + 'static {}
+
+// For our own convinience, let's implement Component for primitives in tests.
+// It will eventually be removed, once tests are not using them anymore.
+// Consider those impls deprecated.
```

Comment:
Can we create an issue for this (if/when) this PR gets merged?

Reply:
Sure, it's a good follow-up once those changes are accepted. Unless we decide to get rid of that in this PR anyway :)